airflow/contrib/hooks/bigquery_hook.py

```diff
@@ -373,6 +373,15 @@ def run_load(self,
         :param field_delimiter: The delimiter to use when loading from a CSV.
         :type field_delimiter: string
         """
+
+        # bigquery only allows certain source formats
+        # we check to make sure the passed source format is valid
+        # if it's not, we raise a ValueError
+        # Refer to this link for more details: https://cloud.google.com/bigquery/docs/reference/rest/v2/jobs#configuration.query.tableDefinitions.(key).sourceFormat
+        allowed_formats = ["CSV", "NEWLINE_DELIMITED_JSON", "AVRO", "GOOGLE_SHEETS"]
+        if source_format.upper() not in allowed_formats:
```

Comment:
Curious, does BQ allow `csv` to be the same as `CSV`? If not, should not do `.upper()` here